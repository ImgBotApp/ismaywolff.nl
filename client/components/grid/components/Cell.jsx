import styled from 'styled-components'

const processGutter = props => props.gutter ? `calc(0.5 * ${props.gutter})` : '0'
const processWidth = props => props.width ? `calc(100% * ${props.width})` : '100%'

const Cell = styled.div`
  flex-basis: auto;
  width: ${processWidth};
  padding: ${processGutter};
`

export default Cell
