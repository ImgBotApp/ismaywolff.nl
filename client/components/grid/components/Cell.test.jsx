import React from 'react'
import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Cell from './Cell'

describe('<Cell />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Cell />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('renders correctly with gutter', () => {
    const wrapper = shallow(<Cell gutter={'10px'} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })

  it('renders correctly with width', () => {
    const wrapper = shallow(<Cell width={1 / 2} />)
    expect(shallowToJson(wrapper)).toMatchSnapshot()
  })
})
