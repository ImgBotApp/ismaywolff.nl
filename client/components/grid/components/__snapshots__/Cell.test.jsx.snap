exports[`<Cell /> renders correctly 1`] = `
<div
  className="WHVWL" />
`;

exports[`<Cell /> renders correctly with gutter 1`] = `
<div
  className="gvmVWP" />
`;

exports[`<Cell /> renders correctly with width 1`] = `
<div
  className="bEtZKN"
  width={0.5} />
`;
